### PR TITLE
[networks] Check for `sockfd_lookup_light` symbol

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Check for `sockfd_lookup_light` symbol to prevent system-probe from crashing when the kprobe is not available. This is a static function that might be inlined in certain builds (so far we've seen this at least in GKE COS)

Related to https://github.com/DataDog/datadog-agent/pull/8524

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/network/config/config_linux.go [hunk 1]
```diff
@@ -37,10 +38,12 @@ func (c *Config) EnabledProbes(runtimeTracer bool) (map[probes.ProbeName]struct{
 			enabled[probes.TCPRetransmit] = struct{}{}
 		}
 
-		enabled[probes.SockFDLookup] = struct{}{}
-		enabled[probes.SockFDLookupRet] = struct{}{}
-		enabled[probes.DoSendfile] = struct{}{}
-		enabled[probes.DoSendfileRet] = struct{}{}
+		if ebpf.VerifyKernelFuncs("sockfd_lookup_light") {
```
Comment (hmahmood) [inline pkg/network/config/config_linux.go:41]: I guess this mean our `sendfile` fix will be no more on GCP?

**Review by hmahmood — APPROVED** [review]

File: pkg/network/config/config_linux.go (hunk 1 above)
Comment (brycekahle) [reply pkg/network/config/config_linux.go:41]: For now. There isn't an easy/fast fix. We are going to assess options for 7.31.

**Review by brycekahle — APPROVED** [review]

File: pkg/network/config/config_linux.go [hunk 2]
```diff
@@ -37,10 +40,12 @@ func (c *Config) EnabledProbes(runtimeTracer bool) (map[probes.ProbeName]struct{
 			enabled[probes.TCPRetransmit] = struct{}{}
 		}
 
-		enabled[probes.SockFDLookup] = struct{}{}
-		enabled[probes.SockFDLookupRet] = struct{}{}
-		enabled[probes.DoSendfile] = struct{}{}
-		enabled[probes.DoSendfileRet] = struct{}{}
+		if matches, _ := ebpf.VerifyKernelFuncs(filepath.Join(c.ProcRoot, "kallsyms"), []string{"sockfd_lookup_light"}); len(matches) > 0 {
```
Comment (brycekahle) [inline pkg/network/config/config_linux.go:43]: This condition is backwards. The function returns the *missing* functions.

Comment (p-lambert) [reply pkg/network/config/config_linux.go:43]: argh. good catch 👍

**Pull request merged**